pkg/gadgets/trace/open/tracer/bpf/opensnoop.bpf.c

```diff
@@ -74,6 +74,7 @@ int ig_open_e(struct trace_event_raw_sys_enter* ctx)
 		struct args_t args = {};
 		args.fname = (const char *)ctx->args[0];
 		args.flags = (int)ctx->args[1];
+		args.mode = (__u16)ctx->args[2];
```

Comment:
What about something like this to avoid having to modify things in two places:

```c
static int handle_enter(const char *fname, int flags, __u16 mode)
{
	struct args_t args = {};
	u64 id = bpf_get_current_pid_tgid();
	/* use kernel terminology here for tgid/pid: */
	u32 tgid = id >> 32;
	u32 pid = id;

	/* store arg info for later lookup */
	if (trace_allowed(tgid, pid)) {
		args.fname = fname;
		args.flags = flags;
		args.mode = mode;
		bpf_map_update_elem(&start, &pid, &args, 0);
	}
	return 0;
}

SEC("tracepoint/syscalls/sys_enter_open")
int ig_open_e(struct trace_event_raw_sys_enter* ctx)
{
	handle_enter((const char *)ctx->args[0], (int)ctx->args[1],
		     (__u16)ctx->args[2]);
}

SEC("tracepoint/syscalls/sys_enter_openat")
int ig_openat_e(struct trace_event_raw_sys_enter* ctx)
{
	handle_enter((const char *)ctx->args[1], (int)ctx->args[2],
		     (__u16)ctx->args[3]);
}
```

NOTE: I didn't test it.